app/src/main/database/index.ts

```diff
@@ -547,6 +559,25 @@ export class DB {
     return Boolean(value);
   }
 
+  getSource(uuid: string): Source | null {
+    const row = this.selectSourceById.get(uuid);
+    if (!row) {
+      return null;
+    }
+    return {
+      uuid: row.uuid,
+      data: JSON.parse(row.data) as SourceMetadata,
+      isRead: this.isTruthy(row.is_seen),
+      hasAttachment: this.isTruthy(row.has_attachment),
+      messagePreview: row.last_message_kind
+        ? {
+            kind: row.last_message_kind,
```

Comment:
Sorry one more when I was testing this behavior against the old client, if kind is file, and we've already fetched + decrypted it, then we should have the plaintext be the base filename. I think/hope that should be straightforward to add?